mllib/src/main/scala/org/apache/spark/ml/source/image/ImageFileFormat.scala

```diff
@@ -98,3 +103,163 @@ private[image] class ImageFileFormat extends FileFormat with DataSourceRegister
     }
   }
 }
+
+object ImageFileFormat {
+
+  val undefinedImageType = "Undefined"
+
+  /**
+   * (Scala-specific) OpenCV type mapping supported
+   */
+  val ocvTypes: Map[String, Int] = Map(
+    undefinedImageType -> -1,
+    "CV_8U" -> 0, "CV_8UC1" -> 0, "CV_8UC3" -> 16, "CV_8UC4" -> 24
+  )
+
+  /**
+   * (Java-specific) OpenCV type mapping supported
+   */
+  val javaOcvTypes: java.util.Map[String, Int] = ocvTypes.asJava
+
+  /**
+   * Schema for the image column: Row(String, Int, Int, Int, Int, Array[Byte])
+   */
+  private[image] val columnSchema = StructType(
+    StructField("origin", StringType, true) ::
+    StructField("height", IntegerType, true) ::
+    StructField("width", IntegerType, true) ::
+    StructField("nChannels", IntegerType, true) ::
+    // OpenCV-compatible type: CV_8UC3 in most cases
+    StructField("mode", IntegerType, true) ::
```

Comment:
I change all column `nullable` attributes to be true, because spark datasource will ignore the nullable attributes specified here and set all column nullable to be true.
So that `ImageFileFormat.schema` will be consistent with `spark.read.format("image").load(...).schema`. There's a python test which check this consistency.
CC @cloud-fan

Reply:
That's not true in structured streaming. Shall we don't change this here for now? Sounds like orthogonal with this change.